core/network/src/p2p/gossip/gossip.rs

```diff
@@ -0,0 +1,321 @@
+                    if let Err(e) = self.execute().await {
+                        error!("Failed to Gossip : {:?}", e);
+                        //ToDo
+
+                    }
+                }
+                _ = self.stop_rx.recv() => {
+                    debug!("Shutting down gossip");
+                    break;
+                }
+            }
+        }
+    }
+
+    async fn execute(&mut self) -> Result<(), Box<dyn Error>> {
+        //ToDo Dummy vec<u8> for now.
+        let bloom = Vec::new();
+
+        let mut request = PullGossipRequest {
+            filter: bloom,
+            salt: Id::default().to_vec(), //ToDo Use default for now
+        };
+
+        let mut msg_bytes = vec![];
+
+        // We scope the lock here to avoid issue later on
+        {
+            let set_guard = self.set.lock().await;
+
+            let elem = set_guard.fetch_elements();
+
+            let filter = elem.serialize().expect("Issue serializing elem");
+            request.filter = filter;
+            request.encode(&mut msg_bytes)?;
+        }
+
+        for _ in 0..self.config.poll_size {
+            {
+                let set = Arc::clone(&self.set.clone());
+
+                // Initialize the callback that will be used upon receiving a response from our gossip attempt
+                let on_response: AppResponseCallback = Arc::new({
+                    move |response_bytes| {
+                        let response = match PullGossipResponse::decode(response_bytes.as_slice()) {
+                            Ok(res) => {
+                                res
+                            }
+                            Err(e) => {
+                                error!("failed to unmarshal gossip response, error: {:?}", e);
+                                return;
+                            }
+                        };
+
+                        // We iterate over the response's gossip
+                        debug!("There are {} gossips", response.gossip.len());
+                        for bytes in response.gossip.iter() {
+                            let mut gossipable: S::Item = S::Item::default();
+                            gossipable.deserialize(bytes).unwrap();
+
+                            let hash = gossipable.get_id();
+
+                            let mut set_guard = set.try_lock().expect("Failed to acquire lock on set");
+                            if let Err(e) = set_guard.add(gossipable) {
+                                error!(
+                            "failed to add gossip to the known set, id: {:?}, error: {:?}"
+                            , hash, e
+                        );
+                                continue;
+                            }
+                        }
+                    }
+                });
+
+                let mut guard = self.client.try_lock().expect("Failed to acquire a lock on client");
+                guard.app_request_any(msg_bytes.clone(), on_response).await;
```

Comment:
I don't think you need to clone `msg_bytes` here. 
```suggestion
                guard.app_request_any(msg_bytes, on_response).await;
```